### DISCUSSION TITLE
Pet recognition

### DISCUSSION THREAD

**Discussion by laurenspriem** [ente-io/ente#2167] [Enhancements]
Automatically recognizing and grouping different pets, including the ability to name them for easy search.

**Comment by haritvsmurali** [comment]
Yes! recognizing faces and tagging it with names for easy search.

**Comment by wwahammy** [comment]
As a someone whose Yorkie daughter passed away last summer, Ente not recognizing her is perhaps the biggest thing missing compared to Google photos.

I wouldn't say machine learning is my area of expertise by any means but I'm willing to give it a try contributing: both for other users... and Penelope: :slightly_smiling_face:

![PXL_20240502_223056132 RAW-01 MP COVER](https://github.com/user-attachments/assets/4e3281eb-70a9-4d46-9440-2f52b6a59046)

**Comment by KestrelWild** [comment]
Please add this. Definitely one of the things I am missing most from Google Photos.

**Comment by but-whythough** [comment]
I'd really love to see this feature

**Comment by primaleph** [comment]
I'm also missing this feature, as someone who recently moved from Google Photos. I've had 4 cats, and the ability to have Ente recognize them would be wonderful.

**Comment by kowtropic** [comment]
Agreed, this is the only thing I'm missing from Google photos

**Comment by primaleph** [comment]
I may have found some open source models that you could use or adapt. As
long as you don't mind a project that uses a MIT license or Apache
license.

https://github.com/blakeblackshear/frigate

https://levelup.gitconnected.com/detect-your-cats-face-with-opencv-21893210737e

https://github.com/GuillaumeMougeot/DogFaceNet

https://www.kaggle.com/datasets/wutheringwang/dog-face-recognition/data

https://github.com/MarQuisCheshire/Pets-Face-Recognition

Many of these links came from a discussion on the same issue, from users of
an app called Immich. In case reading that thread would be useful, here it
is:

[immich-app/immich#7151](https://github.com/immich-app/immich/discussions/7151)

**Comment by lflender** [comment]
Any news on this? Ente not recognizing any face for most of my pet photos is killing me. Please either improve pet detection or give us the possibility to manually tag them.

Cheers.

**Reply by lflender** [reply]
What is weird is that if I search for "dog" or "spider" in my gallery, it will find plenty of results. However, almost all of them are not recognized as faces so I cannot add them to my pet profile.

**Reply by laurenspriem** [reply]
If at least one face of your pet is recognized then you could actually use manual tagging to add the rest of the photos already. It's a workaround for now, but it might work for you

**Comment by sumassemblyreqd** [comment]
Another +1 from me on this. I just switched to Ente and was appalled to find that I couldn't name my pets! What's going on here? This should have been included as a feature when you could name people.

**Comment by breezy-codes** [comment]
Is there any timeline on when we will get this?

**Reply by laurenspriem** [reply]
No timeline, but very high on our list!

**Reply by wwahammy** [reply]
@laurenspriem that's wonderful to hear. What can we as the community do to help this along? You've got a lot of nerds here that care about their furry friends :smile:

**Reply by laurenspriem** [reply]
The challenging part is in getting a good face embedding model, that can analyze the pet facial features. There's no such open source model available, so we're looking to train our own. The hardest part of that is getting the right data: images of pets that are labeled with their identity, so the model can learn to separate different identities. There are some small datasets like this available online, but it's not a lot, so we might need to scrape from public sites that allow for it (like pet adoption sites). We had also considered crowd-sourcing the data within our community, but it'll be too many images and too much effort to coordinate effectively. 

So right now, we're experimenting with the few available open datasets, seeing if we can bootstrap our way there. If you know of any datasets that we could use, please do let us know!

**Comment by CeVague** [comment]
+1 for me to... and don't forget people with exotic pets ! Like ginea pig or rabbit.
I'm not saying you should recognize all of them perfectly, but at least the category for the type of pet could be a great deal !

**Comment by ITechGeek81** [comment]
How about letting people voluntarily submit their own pet pics along with the name of the pet?

**Comment by abby-d** [comment]
Adding my support for this. As silly as it may seem, auto-adding pets to albums shared with my partner is my most-used feature across photos apps. Currently, this is really hit or miss with Ente. If Ente has recognized a pet as having a face, I'm able to then tag future photos with that pet's face. This is almost always a manual process .But one of our pets (a very fuzzy cat boy) hasn't been recognized yet, so I can't even manually add his face. 

My request condensed:

- Allow manual adding of faces so that they can be manually tagged on photos.
- Improve pet recognition. I'm happy to provide a ridiculous amount of cat photos to support the cause.

**Comment by lflender** [comment]
Yes, we definitely need manual tagging, even when the recognition will work fine. I can provide plenty of dog photos, but anyway we will also want exotic pets management, like spiders or snakes, so keep that in mind please. Google Photos already recognizes spiders.

**Reply by breezy-codes** [reply]
I have geckos so I can assist with that exotic pet part!

**Comment by laurenspriem** [comment]
Just a quick update here for everyone that commented on this feature request recently. Pet recognition is now being actively worked on, and we're making good progress. So you can expect an app update on this soon!

**Comment by mizzyc** [comment]
I know this was already deployed, and it is already a great feature that helped me a lot! 

I already have my dog tagged, and I don't know if anyone else is experiencing this, but most of the photos I have to tag her manually, besides being suggested to merge a lot of humans with my dog "person". 

Is there any improvement being worked on for better pet recognition?

**Reply by laurenspriem** [reply]
I assume you are self-hosting or building from source?

Just to clarify: pet recognition is still under internal, active development and not yet out to normal users till further polishing is done.

**Reply by mizzyc** [reply]
No, I'm using Ente's cloud service.

I understand it is still under development, just thought it was a feedback that would worth bringing here! :)

**Comment by siwyd** [comment]
@laurenspriem I was wondering if you have any update perhaps on the pets tagging feature progress? I've been testing and really liking the hosted service, I think this might be the one feature that I'm 'missing' so far.

**Reply by laurenspriem** [reply]
It's in-progress. It got delayed a bit due to other factors, but I still fully expect it to be released before the end of this calendar year.
